Monte_carlo.cpp

```diff
@@ -0,0 +1,324 @@
+            for (int axis=0; axis<3; ++axis) {
+                for (int side=0; side<2; ++side) {
+
+                    // r is variable that contains the distance
+                    // along the direction vector to the boundary being tested.
+                    r = distance_to_cell_edge[axis][side]
+                        / neutron.getDirection(axis);
+                    if (r > 0 & r < tempd) {
+                        tempd = r;
+                        cell_lim_bound.clear();
+                        cell_lim_bound.push_back(axis*2+side);
+                    }
+                    else if (r == tempd) {
+                        cell_lim_bound.push_back(axis*2+side);
+                    }
+                }
+            }
+
+            // move neutron
+            neutron.move(tempd);
+
+            // add distance to cell flux
+            mesh.fluxAdd(cell, tempd, group);
+
+            // determine boundary status
+            for (int sur_side=0; sur_side <6; ++sur_side) {
+                int axis = sur_side/2;
+                int side = sur_side%2;
+
+                // if sur_side is in cell_lim_bound
+                if (std::find(cell_lim_bound.begin(),
+                            cell_lim_bound.end(),sur_side)
+                        != cell_lim_bound.end()) {
+                    if (cell_mins[axis] == bounds.getSurfaceCoord(axis, side)
+                            | cell_maxes[axis] == 
+                            bounds.getSurfaceCoord(axis, side)) {
+                        box_lim_bound.push_back(sur_side);
+                    }
+                }
+            }
+
+            // check boundary conditions on all hit surfaces
+            for (int sur_side=0; sur_side <6; ++sur_side) {
+                int axis = sur_side/2;
+                int side = sur_side%2;
+
+                // if sur_side is in box_lim_bound
+                if (std::find(box_lim_bound.begin(),
+                            box_lim_bound.end(),sur_side)
+                        != box_lim_bound.end()) {
+
+                    // if the neutron is reflected
+                    if (bounds.getSurfaceType(axis, side) == 1) {
+                        neutron.reflect(axis);
+
+                        // place neutron on boundary to eliminate 
+                        //    roundoff error
+                        bound_val = bounds.getSurfaceCoord(axis, side);
+                        neutron.setPosition(axis, bound_val);
+                    }
+
+                    // if the neutron escapes
+                    if (bounds.getSurfaceType(axis, side) == 0) {
+                        neutron.kill();
+                        neutron_distance = tempd;
+                        tallies[LEAKS].add(1);
+                    }
+                }
+            }
+ 
+            // shorten neutron distance to collision
+            neutron_distance -= tempd;
+
+            // get new neutron cell
+            if (neutron_distance != 0.0) {
```

Comment:
I would strongly suggest changing this to ``neutron_distance > 0.0``. With roundoff error, the chances of arriving at exactly 0.0 are slim.